File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    arr = []
    cnt = 1
    for i in range(len(progresses)):
        arr.append(math.ceil((100-progresses[i])/speeds[i]))
    
    max_num = arr[0]
    for i in range(1, len(arr)):
        if arr[i] <= max_num:
            cnt += 1
        else:
            answer.append(cnt)
            max_num = arr[i]
            cnt = 1
    answer.append(cnt)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> list = new ArrayList<>();
        
        Deque<int[]> q = new ArrayDeque<>();
        
        for(int i = 0; i<progresses.length; i++){
            q.add(new int[]{progresses[i], speeds[i]});
        }
        
        while(!q.isEmpty()){
            
            int size = q.size();
            for(int i = 0; i<size; i++){
                int[] work = q.poll();
                q.add(new int[]{work[0]+work[1], work[1]});
            }
            
            int cnt = 0;
            while(!q.isEmpty() && q.peekFirst()[0] >= 100){
                q.poll();
                cnt++;
            }
            
            if(cnt > 0){
                list.add(cnt);
            }
            
        }
        
        int[] answer = new int[list.size()];
        
        for(int i = 0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
}